tests/util/test_unit_conversion.py

```diff
@@ -102,95 +102,11 @@
-        # that we don't miss any in the tests.
-        obj
-        for _, obj in inspect.getmembers(unit_conversion)
-        if inspect.isclass(obj)
-        and issubclass(obj, BaseUnitConverter)
-        and obj != BaseUnitConverter
-    ],
-)
-def test_all_converters(converter: type[BaseUnitConverter]) -> None:
-    """Ensure all unit converters are tested."""
-    assert converter in _ALL_CONVERTERS, "converter is not present in _ALL_CONVERTERS"
-    assert converter in _GET_UNIT_RATIO, "converter is not present in _GET_UNIT_RATIO"
-
-
-@pytest.mark.parametrize(
-    "converter,valid_unit",
-    [
-        # Ensure all units are tested
-        (converter, valid_unit)
-        for converter, valid_units in _ALL_CONVERTERS.items()
-        for valid_unit in valid_units
-    ],
-)
-def test_convert_same_unit(converter: type[BaseUnitConverter], valid_unit: str) -> None:
-    """Test conversion from any valid unit to same unit."""
-    assert converter.convert(2, valid_unit, valid_unit) == 2
-
-
-@pytest.mark.parametrize(
-    "converter,valid_unit",
-    [
-        # Ensure all units are tested
-        (converter, valid_unit)
-        for converter, valid_units in _ALL_CONVERTERS.items()
-        for valid_unit in valid_units
-    ],
-)
-def test_convert_invalid_unit(
-    converter: type[BaseUnitConverter], valid_unit: str
-) -> None:
-    """Test exception is thrown for invalid units."""
-    with pytest.raises(HomeAssistantError, match="is not a recognized .* unit"):
-        converter.convert(5, INVALID_SYMBOL, valid_unit)
-
-    with pytest.raises(HomeAssistantError, match="is not a recognized .* unit"):
-        converter.convert(5, valid_unit, INVALID_SYMBOL)
-
-
-@pytest.mark.parametrize(
-    "converter,from_unit,to_unit",
-    [
-        # Pick any two units
-        (converter, valid_units[0], valid_units[1])
-        for converter, valid_units in _ALL_CONVERTERS.items()
-    ],
-)
-def test_convert_nonnumeric_value(
-    converter: type[BaseUnitConverter], from_unit: str, to_unit: str
-) -> None:
-    """Test exception is thrown for nonnumeric type."""
-    with pytest.raises(TypeError):
-        converter.convert("a", from_unit, to_unit)
-
-
-@pytest.mark.parametrize(
-    "converter,from_unit,to_unit,expected",
-    [
-        (converter, item[0], item[1], item[2])
-        for converter, item in _GET_UNIT_RATIO.items()
-    ],
-)
-def test_get_unit_ratio(
-    converter: type[BaseUnitConverter], from_unit: str, to_unit: str, expected: float
-) -> None:
-    """Test unit ratio."""
-    ratio = converter.get_unit_ratio(from_unit, to_unit)
-    assert ratio == pytest.approx(expected)
-    assert converter.get_unit_ratio(to_unit, from_unit) == pytest.approx(1 / ratio)
-
-
-@pytest.mark.parametrize(
-    "value,from_unit,expected,to_unit",
-    [
+# Dict containing a conversion test for every know unit.
+_CONVERTED_VALUE: dict[
+    type[BaseUnitConverter], list[tuple[float, str | None, float, str | None]]
+] = {
```

Comment:
This is the first half of the PR: merged the parametrization lists into a single dictionary.